docs/architecture.md

```diff
@@ -0,0 +1,52 @@
+# Architecture
+
+Newman Dashboard is a real-time reporter for newman to control, view and debug runs on a WebUI dashboard.
+
+## Pub - Sub
+
+We establish a pub/sub architecture between the newman runs and the WebUI(frontend) through a central server known as the broker. <br/>
+The frontend and the runs never know about one another directly, they just connect to the broker as clients.<br/>
+The broker pushes each of these clients into separate "rooms" to maintain abstraction and isolation between them i.e. every client should receive messages concerned with itself and no other client. <br/>
+
+## Flow
+
+### Connecting a newman run to broker
+
+For connecting a newman run to the broker there are 2 prerequisites:
+
+1. Dashboard should be started on `http://localhost:5001/` using the newman-dashboard CLI
+2. `newman-dashboard` should have been provided as a reporter for that newman run.
+
+```
+newman run <collection> -r dashboard
+```
+
+1. When this newman run will execute, the reporter will try to connect to the broker at `http://localhost:5001/`.
+2. For this connection to be authenticated, it would need to provide 2 things during the socket handshake:
+    - A unique id to solely identify this run.
+    - The type of this client, in this case `newman-run`
```

Comment:
We don't need to provide the implementation details in the architecture doc. We can have separate docs for Newman integration and frontend integration.

Reply:
I have used this to provide more context about the run id and the concept of rooms. Do you want to remove this? All the other changes have been made! 🚀

Reply:
I don't think it should be in the architecture document. If we decide to change the authentication later, this document should not need to be changed because the architecture is still the same. Thoughts?

Reply:
Okay got it! I have removed this!